doc/dev_guide/internals/box_protocol.rst

```diff
@@ -706,6 +708,96 @@ IPROTO_SQL_TEXT (0x40) and statement-text (string) if executing an SQL string.
 Thus the IPROTO_PREPARE map item is the same as the first item of the
 :ref:`IPROTO_EXECUTE <box_protocol-execute>` body.
 
+..  _box_protocol-begin:
+
+IPROTO_BEGIN = 0x0e
+~~~~~~~~~~~~~~~~~~~~~
+
+Begin a transaction in the specified stream.
+See :ref:`stream:begin() <net_box-stream_begin>`.
+The body is optional and can contain two items:
+
+..  cssclass:: highlight
+..  parsed-literal::
+
+    # <size>
+    msgpack(:samp:`{{MP_UINT unsigned integer = size(<header>) + size(<body>)}}`)
+    # <header>
+    msgpack({
+        IPROTO_REQUEST_TYPE: IPROTO_BEGIN,
+        IPROTO_SYNC: :samp:`{{MP_UINT unsigned integer}}`,
+        IPROTO_STREAM_ID: :samp:`{{MP_UINT unsigned integer}}`
+    })
+    # <body>
+    msgpack({
+        IPROTO_TIMEOUT: :samp:`{{MP_DOUBLE}}`,
+        IPROTO_TXN_ISOLATION: :samp:`{{MP_UINT unsigned integer}}`
+    })
+
+IPROTO_TIMEOUT is an optional timeout (in seconds). After it expires,
+the transaction will be rolled back automatically.
+
+.. // TODO: add link to transaction isolation docs once they're ready
+IPROTO_TXN_ISOLATION is the transaction isolation level. It can take
+the following values:
+
+.. // TODO: provide links to level descriptions
+- ``TXN_ISOLATION_DEFAULT = 0``	-- use the global default level (default value)
+- ``TXN_ISOLATION_READ_COMMITTED = 1`` -- read changes that are committed but not confirmed yet
+- ``TXN_ISOLATION_READ_CONFIRMED = 2`` -- read confirmed changes
+- ``TXN_ISOLATION_BEST_EFFORT = 3`` -- determine isolation level automatically
```

Comment:
```suggestion
- ``TXN_ISOLATION_DEFAULT = 0``	-- use the global default level (default value).
- ``TXN_ISOLATION_READ_COMMITTED = 1`` -- read changes that are committed but not confirmed yet.
- ``TXN_ISOLATION_READ_CONFIRMED = 2`` -- read confirmed changes.
- ``TXN_ISOLATION_BEST_EFFORT = 3`` -- determine isolation level automatically.
```
You can find the list punctuation rules in our [style guide](https://www.tarantool.io/en/doc/latest/contributing/docs/style/#lists).